widget/embedded/src/components/SwapDetails/SwapDetails.helpers.tsx

```diff
@@ -0,0 +1,215 @@
+                  pendingSwap.inputAmount,
+                  AMOUNT_DECIMALS,
+                  AMOUNT_DECIMALS
+                )
+              : numberToString(
+                  pendingSwap.steps[index - 1].outputAmount,
+                  AMOUNT_DECIMALS,
+                  AMOUNT_DECIMALS
+                ),
+        },
+      },
+      to: {
+        token: { displayName: step.toSymbol, image: step.toLogo },
+        chain: {
+          displayName: step.toBlockchain,
+          image: step.toBlockchainLogo,
+        },
+        price: {
+          value: step.outputAmount
+            ? numberToString(
+                step.outputAmount,
+                AMOUNT_DECIMALS,
+                AMOUNT_DECIMALS
+              )
+            : numberToString(
+                step.expectedOutputAmountHumanReadable,
+                AMOUNT_DECIMALS,
+                AMOUNT_DECIMALS
+              ),
+        },
+      },
+      swapper: { displayName: step.swapperId, image: step.swapperLogo },
+      alerts: (
+        <Alerts>
+          {step.explorerUrl?.map((explorerUrl, index, urls) => {
+            const key = index + explorerUrl.url;
+            return (
+              <Fragment key={key}>
+                <Alert
+                  type={
+                    (index === urls.length - 1 &&
+                      step.status === 'running' &&
+                      (index !== 0 ||
+                        (index === 0 &&
+                          explorerUrl.description !== 'Approve'))) ||
+                    (step.status === 'waitingForApproval' &&
+                      !isNetworkStatusInWarningState(step))
+                      ? 'loading'
+                      : index === urls.length - 1 &&
+                        (index !== 0 ||
+                          (index === 0 &&
+                            explorerUrl.description !== 'Approve')) &&
+                        step.status === 'failed'
+                      ? 'error'
+                      : 'success'
+                  }
+                  title={
+                    !explorerUrl.description
+                      ? i18n.t('View transaction')
+                      : `${explorerUrl.description
+                          .substring(0, 1)
+                          .toUpperCase()}${explorerUrl.description.substring(
+                          1
+                        )} Tx`
+                  }
+                  action={
+                    explorerUrl.url && (
+                      <Button
+                        variant="ghost"
+                        dense
+                        onClick={() => window.open(explorerUrl.url, '_blank')}>
+                        <LinkIcon size={16} />
+                      </Button>
+                    )
+                  }
+                />
+              </Fragment>
+            );
+          })}
+          {step.status !== 'failed' && isNetworkStatusInWarningState(step) && (
+            <Alert
```

Comment:
I think we could separate `WaitingForNetworkChange` and `WaitingForConnectingWallet` alerts. (or refactor current code)

Reply:
fixed.